/*******************************************************************************
 * MICROSOFT CONFIDENTIAL & PROPRIETARY
 *
 * Copyright (c) 2019 Microsoft Corp.
 * All Rights Reserved.
 ******************************************************************************/

syntax = "proto3";

package SwitchML;

service RDMAServer {
  rpc RDMAConnect (RDMAConnectRequest) returns (RDMAConnectResponse) {}
}

message RDMAConnectRequest {
  uint64 job_id = 1;
  int32 my_rank = 2;
  int32 job_size = 3;
  uint64 mac = 4;
  uint32 ipv4 = 5;
  uint32 rkey = 6;
  enum PacketSize {
    IBV_MTU_128  = 0;
    IBV_MTU_256  = 1;
    IBV_MTU_512  = 2;
    IBV_MTU_1024 = 3;
  }
  PacketSize packet_size = 7;
  uint32 message_size = 8;
  repeated uint32 qpns = 9;
  repeated uint32 psns = 10;
}

message RDMAConnectResponse {
  uint64 job_id = 1;
  repeated uint64 macs = 2;
  repeated uint32 ipv4s = 3;
  repeated uint32 rkeys = 4;
  repeated uint32 qpns = 5;
  repeated uint32 psns = 6;
}


